### PR TITLE
Add alert for deprecated apis

### PR DESCRIPTION
**Pull request by tnozicka**
/cc @soltysh 

/hold
(just in case, I think this should fail on current offenders)

### CODE REVIEW

**Comment by tnozicka** [comment]
I have manually checked that the alerts are present in the prometheus dump from the CI run - that should have failed aws-e2e - monitoring team is working on a fix. We should merge when the check is fixed, we get really green e2e and no alerts are firing. (Also means making the offenders to fix their code.)

**Comment by tnozicka** [comment]
e2e fix in https://github.com/openshift/origin/pull/24005

File: manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml [hunk 1]
```diff
@@ -35,3 +35,34 @@ spec:
...
+        message: A client in the cluster is using deprecated apps/v1beta2 API that will be removed soon.
+      expr: |
+        apiserver_request_count{group="apps",version="v1beta2"}
+      labels:
+        severity: warning
+    - alert: UsingDeprecatedAPIExtensionsV1Beta1
+      annotations:
+        message: A client in the cluster is using deprecated extensions/v1beta1 API that will be removed soon.
+      expr: |
+        apiserver_request_count{group="extensions",version=~"v1beta1",resource!~"(?:ingresses|)",client!~".*/kube-controller-manager"}
```
Comment (soltysh) [inline manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: Any particular reason you do `=~` when matching version here? I doubt it's required.

Comment (tnozicka) [reply manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: not really, just wanted to set pattern for easily adding another one if need be

Comment (tnozicka) [reply manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: oh, you mean version - that's a leftover from where it was matching v1beta2 as well before I made it separate alert, will fix it tomorrow

**Comment by sttts** [comment]
:+1:

LGTM

File: manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml (hunk 1 above)
Comment (tnozicka) [reply manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: fixed

**Comment by tnozicka** [comment]
alerts are firing in https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/625/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws/2869/artifacts/e2e-aws/metrics/prometheus.tar , the e2e that should stop merging it is green :(

**Comment by tnozicka** [comment]
ok, this needs to land for feature

/hold cancel

I'll follow up with a high prio BZ to fix the e2e check; the ones for offenders are already filled

**Comment by tnozicka** [comment]
ref https://bugzilla.redhat.com/show_bug.cgi?id=1766638

**Pull request merged**